src/integration-test/java/com/commercetools/sync/integration/commons/utils/CartDiscountITUtils.java

```diff
@@ -0,0 +1,150 @@
+                                .description(CART_DISCOUNT_DESC_1)
+                                .validFrom(JANUARY_FROM)
+                                .validUntil(JANUARY_UNTIL)
+                                .build();
+
+    public static final CartDiscountDraft CART_DISCOUNT_DRAFT_2 =
+        CartDiscountDraftBuilder.of(LocalizedString.of(Locale.ENGLISH, CART_DISCOUNT_KEY_2),
+            CART_DISCOUNT_CART_PREDICATE_2,
+            CART_DISCOUNT_VALUE_2,
+            CART_DISCOUNT_TARGET_2,
+            SORT_ORDER_2,
+            false)
+                                .active(false)
+                                .description(CART_DISCOUNT_DESC_2)
+                                .validFrom(FEBRUARY_FROM)
+                                .validUntil(FEBRUARY_UNTIL)
+                                .build();
+
+
+    /**
+     * Deletes all cart discounts from CTP project, represented by provided {@code ctpClient}.
+     *
+     * @param ctpClient represents the CTP project the cart discounts will be deleted from.
+     */
+    public static void deleteCartDiscounts(@Nonnull final SphereClient ctpClient) {
+        queryAndExecute(ctpClient, CartDiscountQuery.of(), CartDiscountDeleteCommand::of);
+    }
+
+    /**
+     * Deletes all cart discounts from CTP projects defined by {@code CTP_SOURCE_CLIENT} and {@code CTP_TARGET_CLIENT}.
+     */
+    public static void deleteCartDiscountsFromTargetAndSource() {
+        deleteCartDiscounts(CTP_SOURCE_CLIENT);
+        deleteCartDiscounts(CTP_TARGET_CLIENT);
+    }
+
+
+    public static void populateSourceProject() {
+        CTP_SOURCE_CLIENT.execute(CartDiscountCreateCommand.of(CART_DISCOUNT_DRAFT_1)).toCompletableFuture().join();
+        CTP_SOURCE_CLIENT.execute(CartDiscountCreateCommand.of(CART_DISCOUNT_DRAFT_2)).toCompletableFuture().join();
+    }
+
+    public static void populateTargetProject() {
+        CTP_TARGET_CLIENT.execute(CartDiscountCreateCommand.of(CART_DISCOUNT_DRAFT_1)).toCompletableFuture().join();
+    }
+
+    /**
+     * Tries to fetch cart discount of {@code key} using {@code sphereClient}.
+     *
+     * @param sphereClient sphere client used to execute requests.
+     * @param key          key of requested cart discount.
+     * @return {@link Optional} which may contain type of {@code key}.
+     */
+    public static Optional<CartDiscount> getCartDiscountByKey(
+        @Nonnull final SphereClient sphereClient,
+        @Nonnull final String key) {
+
+        final CartDiscountQuery query = CartDiscountQueryBuilder
+            .of() //todo: SUPPORT-4443 need to be merged from name to key.
+            .plusPredicates(queryModel -> queryModel.name().lang(Locale.ENGLISH).is(key))
+            .build();
+
+        return sphereClient.execute(query).toCompletableFuture().join().head();
+    }
+
+    public static List<String> getSortOrders(int capacity) {
+        // get 100 sortOrder list with odd numbers ex: 0.01, 0.03..
```

Comment:
it's easy to ensure that sort order is not ending with zero.

Reply:
ok then maybe write in the comment as it's not very clear.

Maybe even better as a Javadoc:

```java
    /**
     * Builds a list of odd sortOrder strings that start with `0.01`. They are odd because 
     * because sortOrder in CTP is not allowed to end with a zero and should be decimal value 
     * > 0 and <1.
     *
     * @param capacity the number of sort orders to build.
     */
```

Reply:
nice idea, e257d68